### ISSUE TITLE
Parsing non-.mjs files that are es modules?

### ISSUE THREAD

**Issue by hilary-L** [standard-things/esm#323]
First off, I love this project. You've solved some longstanding issues I've been having and I think have made a pretty cool side project possible. So thank you!

I have a question about basic setup, which may just reflect confusion on my part as to how this is all supposed to work. Have included the code in my little quicky project below, but here's the basic issue: if I run my code with `node index.js` it chokes with an "SyntaxError: 'import' and 'export' may only be used in ES modules" error. If I comment out the lines `const otherReactTest = await import('./src/OtherReactTest.js');` and `console.log('react js test: ', otherReactTest);` it works perfectly.

I'm not totally surprised that this is happening, and indeed it might be how things are intended to work. But, I'm just wondering is there a way to get it to work with that OtherReactTest.js file? Or is it all .mjs files all the time no matter what?

package.json:
```
  "dependencies": {
    "@babel/core": "^7.0.0-beta.42",
    "@babel/polyfill": "^7.0.0-beta.42",
    "@babel/preset-env": "^7.0.0-beta.42",
    "@babel/preset-react": "^7.0.0-beta.42",
    "@babel/preset-stage-3": "^7.0.0-beta.42",
    "@babel/register": "^7.0.0-beta.42",
    "@std/esm": "^0.25.3",
    "react": "^16.2.0"
  },
```

index.js:
```
require = require('@std/esm')(module, {
	cjs: true
});

require("@babel/polyfill");

require('@babel/register')({
	babelrc: false,
	presets: [
		['@babel/preset-env', {loose: true, modules: false}],
		['@babel/preset-react'],
		['@babel/preset-stage-3']
	]
});

module.exports = require('./index.mjs').default
```

index.mjs:
```
const letsDoThis = async () => {
	const reactTestModule = await import('./src/ReactTest.mjs');
	const otherReactTest = await import('./src/OtherReactTest.js');
	const statelessReactTest = await import('./src/statelessReactTest');

	console.log('react mjs test: ', reactTestModule);
	console.log('react js test: ', otherReactTest);
	console.log('stateless react test: ', statelessReactTest);
}

letsDoThis().catch((error) => {
	console.log('error!', error);
});
```

/src/ReactTest.mjs:
```
import React, { Component } from 'react';

class TestComponent extends Component {
	render() {
		return (
			<div>
				<h1>this is pretty cool</h1>
			</div>
		)
	}
}

export default TestComponent
```

/src/OtherReactTest.js
```
import React, { Component } from 'react';

class TestComponent extends Component {
	render() {
		return (
			<div>
				<h1>this is also pretty cool</h1>
			</div>
		)
	}
}

export default TestComponent
```

/src/StatelessReactTest.mjs
```
import React from 'react';

const ThisOtherTestComponent = (props) => {
	return (
		<div>
			<h3>But this is cooler!</h3>
		</div>
	)
}

export default ThisOtherTestComponent
```

**Comment by jdalton** [comment]
Hi @hilary-L!

For use with `@babel/register` you'll want to invoke it via `node -r` so in your package.json create a `"scripts"` entry called `"start"`:
```js
"scripts": {
  "start": "node -r @std/esm -r @babel/register -r @babel/polyfill index.mjs"
}
```
Then
```
npm run start
```

**Comment by hilary-L** [comment]
Oh so it's not possible to do it using the requires in the index.js file then? I'm fine if that's the case -- was just trying to piece together a setup from past closed issues on this repo and saw that method used before.

**Comment by jdalton** [comment]
> Oh so it's not possible to do it using the requires in the index.js file then? I'm fine if that's the case -- was just trying to piece together a setup from past closed issues on this repo and saw that method used before.

Not with `@babel/register`. Because `@babel/register` hooks into the global module loader you'll need to load `@std/esm` first. Since the in package usage of `@std/esm` is isolated from the global loader you need to use `node -r @std/esm` to unlock it.

**Comment by hilary-L** [comment]
Hi again. So, I tried your method, but now it chokes on the JSX in the react components that I'm importing. (I think this is because it's not using the correct babel presets). I tried specifying the presets both via a .babelrc and in the start script you specified up above, and neither worked. Do you know how I might get babel-register to use the presets when I'm using it in a start script like you suggested?

Related to that, but, I saw in #321 that you mentioned that babel-register is not suitable for production use... I've seen that mentioned before. Is this the case for everything *but* an application similar to the quicky one I've posted above? Without going into too much extra detail, I'm working on something that will have a CLI where a user will specify a path to an es module, and that es module will be imported using dynamic imports and processed by my application. This would seem to necessitate using babel-register so that the user can specify the module location either via the command line, or perhaps via a configs file... unless I'm just thinking about this wrong.

Any thoughts you have would be appreciated!

**Comment by jdalton** [comment]
Hi @hilary-L!

For `@babel/register` use and questions for `node -r @babel/register` you can visit their [slack channel](https://slack.babeljs.io/). Once you get `node -r @babel/register` working without `-r std/esm` then you can work `@std/esm` back in. Just a heads up the `@std/esm` package is now [esm](https://www.npmjs.com/package/esm) and ships with different zero config defaults. 

If you'd like you can create a simple repro repo and I can dig in and see if I can get things running.

**Comment by hilary-L** [comment]
I'll pop into the slack channel to see if anyone there has any ideas. But in the meantime, I've pushed things up to a test repo at https://github.com/hilary-L/esm-test-repo. This repo still has the index.js file in it, so you can see the difference in the errors produced by running ``` node index.js ``` and ``` npm run start ``` ...if you happen to figure things out that would be super.

**Comment by hilary-L** [comment]
Hi @jdalton just thought you'd like to know that I think I might have this working... weirdly, though, I don't know why. It's not using the method you suggested -- tried that a bunch and it kept choking. I ended up just fiddling with it for a while and combining different approaches, and now if you run ``` node index.js ``` you'll see the logged statements for all of the various components that have been imported. 

I pushed the latest code to the master branch of the tiny repo I linked above. If you have a moment to take a look and see what I changed that made it work, I'd love to know.
